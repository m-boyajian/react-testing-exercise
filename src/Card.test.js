import React from "react";
import { render } from "@testing-library/react";
import Card from "./Card";
import TEST_IMAGES from "./_testCommon.js";

it("renders without crashing", function() {
    render(<Card />);
});

TEST_IMAGES.forEach((photo, index) => {
  it("matches snapshot", function() {
    const {asFragment} = render(<Card src={photo.src} caption={photo.caption} />);
    expect(asFragment()).toMatchSnapshot();
  });
});