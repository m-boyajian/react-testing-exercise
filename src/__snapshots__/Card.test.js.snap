// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 1
    </h4>
    <img
      alt="testing image 1"
      class="Card-image"
      src="test1.com"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 2`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 2
    </h4>
    <img
      alt="testing image 2"
      class="Card-image"
      src="test2.com"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot 3`] = `
<DocumentFragment>
  <div
    class="Card"
  >
    <h4
      class="Card-title"
    >
      testing image 3
    </h4>
    <img
      alt="testing image 3"
      class="Card-image"
      src="test3.com"
    />
    <small
      class="Card-small"
    >
      Image  of .
    </small>
  </div>
</DocumentFragment>
`;
